C7/PCXToGodot.cs

```diff
@@ -5,65 +5,102 @@
 public class PCXToGodot : Godot.Object
 {
 	private readonly static byte CIV3_TRANSPARENCY_START = 254;
-	
-	public static ImageTexture getImageTextureFromPCX(Pcx pcx) {
-		Image ImgTxtr = ByteArrayToImage(pcx.ColorIndices, pcx.Palette, pcx.Width, pcx.Height);
+
+	public static ImageTexture getImageTextureFromImage(Image image) {
 		ImageTexture Txtr = new ImageTexture();
-		Txtr.CreateFromImage(ImgTxtr, 0);
+		Txtr.CreateFromImage(image, 0);
 		return Txtr;
 	}
 
+	public static ImageTexture getImageTextureFromPCX(Pcx pcx) {
+		Image ImgTxtr = ByteArrayToImage(pcx.ColorIndices, pcx.Palette, pcx.Width, pcx.Height);
+		return getImageTextureFromImage(ImgTxtr);
+	}
+
 	public static ImageTexture getImageTextureFromPCX(Pcx pcx, int leftStart, int topStart, int croppedWidth, int croppedHeight) {
-		Image Image = getImageFromPCX(pcx, leftStart, topStart, croppedWidth, croppedHeight);
-		ImageTexture Txtr = new ImageTexture();
-		Txtr.CreateFromImage(Image, 0);
-		return Txtr;
+		Image image = getImageFromPCX(pcx, leftStart, topStart, croppedWidth, croppedHeight);
+		return getImageTextureFromImage(image);
+	}
+
+	// BufferData should be as large as the largest PCX file needed to be loaded in pixels
+	// For now, I will assume no PCX file is larger than 4k resolution
+	// It's efficient to use a shared BufferData, ColorData, and AlphaData instead of having to reinitialize these arrays for every Image load
+	// However, this will become a problem if (when?) we move towards parallelization
+	// But for now, it's an improvement
+	public static int[] BufferData = new int[3840 * 2160];
+	public static int[] ColorData = new int[256];
+	public static int[] AlphaData = new int[256];
+
+	public static void loadPalette(byte[,] palette, bool shadows = false) {
+		int Red, Green, Blue;
+		int Alpha = 255 << 24;
+
+		for (int i = 0; i < 256; i++) {
+			Red = palette[i, 0];
+			Green = palette[i, 1] << 8;
+			Blue = palette[i, 2] << 16;
+			ColorData[i] = Red + Green + Blue + Alpha;
+		}
+
+		for (int i = CIV3_TRANSPARENCY_START; i < 256; i++) {
+			ColorData[i] &= 0x00ffffff;
+		}
+
+		if (shadows) {
+			for (int i = 240; i < 256; i++) {
+				ColorData[i] = ((255 - i) * 16) << 24;
+			}
+		}
```

Comment:
This looks like a nice improvement to shadow readability to me.  @JimOfLeisure Does this make sense to you, as the author of the shadow code and most knowledgeable person on it?

Reply:
Looks good to me. I wrote that a while back. I seemed to be coding shadows on the fly where here he's building a discrete palette with alpha-channel shadows. That works.